libs/ngx-dsfr/input/src/lib/input.component.html

```diff
@@ -1,49 +1,48 @@
-<div class="fr-form-group">
-  <div class="fr-input-group" [ngClass]="{'fr-input-group--error': hasFailed, 'fr-input-group--valid': hasSucceeded}">
-    <label class="fr-label" [for]="id">
-      {{ label }}
-      <span *ngIf="hint" class="fr-hint-text">
-        {{ hint }}
-      </span>
-    </label>
-    <ng-container *ngIf="type === 'textarea'; else standardInput">
-      <textarea
+<div
+  class="fr-input-group"
+  [ngClass]="inputGroupClasses">
+  <label class="fr-label" [for]="id">
+    {{ label }}
+    <span *ngIf="hint" class="fr-hint-text">
+      {{ hint }}
+    </span>
+  </label>
+  <ng-container *ngIf="type === 'textarea'; else standardInput">
+    <textarea
+      class="fr-input"
+      [name]="name ? name : ''"
+      [id]="id"
+      [placeholder]="placeholder"
+      [attr.aria-describedby]="ariaDescribedby"
+      [disabled]="disabled ? 'disabled' : false"
+      [ngClass]="inputClasses"
+      (blur)="onBlur($event)"
+      (keyUp)="onKeyup($event)"
+      (focus)="onFocus($event)"
+      [(ngModel)]="value">
+    </textarea>
+  </ng-container>
+  <ng-template #standardInput>
+    <div [ngClass]="icon ? 'fr-input-wrap ' + icon : ''">
+      <input
         class="fr-input"
+        [type]="type"
         [name]="name ? name : ''"
         [id]="id"
         [placeholder]="placeholder"
-        [attr.aria-labelledby]="ariaLabelledBy"
-        [disabled]="disabled ? 'disabled' : false"
+        [attr.aria-describedby]="ariaDescribedby"
+        [disabled]="disabled"
         [ngClass]="{'fr-input--error': hasFailed, 'fr-input--valid': hasSucceeded}"
```

Comment:
For consistency, same remark here on how to handle classes depending on Component logic and Input properties